File: Problem2_IsomorphicStrings.java
//Time Complexity : o(n)
//Space Complexity : o(m)
//Did this code successfully run on Leetcode : Yes
//Any problem you faced while coding this : No

class Solution {
    public boolean isIsomorphic(String s, String t) {
        
        
       if(s == null || t == null){
            return false;
        }
        if(s.length() == 0 || t.length() == 0){
            return true;
        }
        
        Map<Character,Character> corr = new HashMap<Character,Character>();
        for(int i = 0; i < s.length(); i ++){
            char sc = s.charAt(i);
            char tc = t.charAt(i);
            if(corr.containsKey(sc)){
                if(corr.get(sc) != tc){
                    return false;
                }
            }
            else{
                if(corr.containsValue(tc)){
                    return false;   
                }
                else{
                    corr.put(sc,tc);
                }
            }
        }
        return true;
    }
}

File: Problem 2 - Isomorphic Strings.py
# Time Complexity :  O(n) number of characters in string
# Space Complexity : O(1) because max number of characters in map will be constant
# Did this code successfully run on Leetcode : yes
# Any problem you faced while coding this : no

class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        hashmap = {}
        hashset = set()
        for i in range(len(s)):
            if s[i] not in hashmap: 
                if t[i] not in hashset:
                    hashset.add(t[i])
                else:
                    return False
                hashmap[s[i]] = t[i]
            else:
                if hashmap[s[i]] != t[i]:
                    return False
        return True